hw/ip/i2c/doc/_index.md

```diff
@@ -243,10 +244,57 @@ Other features may also be required for complete SMBus functionality.)
 ### Clock Stretching
 As described in the I2C specification, a target device can pause a transaction by holding SCL low.
 A target device stretches the clock automatically when a host device reads from the target and the target's TX FIFO is empty.
-The present implementation also supports a mode in which a target device can stretch the clock after completing a transaction, if it needs more time, e.g., to store the acquired data byte in flash memory.
-If {{< regref STRETCH_CTRL.EN >}} is asserted, a target stretches the clock after an ingoing transaction completes.
-However, the target stretches the clock only if it is certain that the next transaction is intended for it and not for another target device on the bus.
-The target device stops stretching the clock and resumes normal operation when {{< regref STRETCH_CTRL.RESUME >}} is asserted.
+Also, a target device stretches the clock automatically when a host device writes to the target and the target's ACQ FIFO is full.
+The present implementation also supports few additional modes in which a target device can stretch the clock after completing a transaction if it needs more time, e.g., to store the acquired data byte in flash memory, etc.
+If {{< regref STRETCH_CTRL.ENABLEADDR >}} is asserted, a target stretches the clock after acquiring an address and making certain that the transaction is intended for it and not for another device on the bus.
+
+{{<wavejson>}}
+{signal: [
+  {name: 'clock', wave: 'p..................'},
+  {name: 'stretch_en_addr', wave: '1..................'},
+  {name: 'stretch_stop', wave: '0.............1....'},
+  {name: 'bit_ack', wave: '0........1.........'},
+  {name: 'address_match', wave: '0........1.........'},
+  {name: 'SCL host driver', wave: '0z.0..z.0..z.0..z.0'},
+  {name: 'SCL target driver', wave: 'z.........0....z...'},
+  {name: 'SCL bus', wave: '0u.0..u.0.......u.0'},
+],
+ head: {text: 'Clock stretching after matching an address (cycle 10), normal operation is resumed and address is acknowledged (cycle 15)'}
```

Comment:
Hey Igor, 
A couple recommended changes here:
1. If clock stretching is enabled, the host isn't going to drop SCL until it can make the clock.  It's a minor detail, but it helps the figure make more sense if we get this right.
2. You can add "tick:1" after your header text to explicitly mark the cycle numbers
3. Making this just one clock longer fixes the cycle problem.

To help clarify this I added a separate PR onto you fork: 
https://github.com/igk8/opentitan/pull/1

You can merge that as you like or just take the pieces and add them directly here, what ever you like. (Just don't forget to pull your fork back down once you do the merge!!)

Reply:
Hi Martin,
Thank you for the review. I have made these changes.